lightningd/plugin.c

```diff
@@ -1055,6 +1022,9 @@ static void plugin_config(struct plugin *plugin)
 	json_add_string(req->stream, "rpc-file", ld->rpc_filename);
 	json_object_end(req->stream);
 
+	if (plugin->signal_startup)
+		json_add_bool(req->stream, "startup", plugin->plugins->startup);
```

Comment:
I don't understand why having this set unconditionally breaks older pylightning?  We should be able to add init fields without repercussions; if we can't, can we fix that now so this won't be a problem in future?

Reply:
This is not a problem with plugins in general, but with how Pylightning (handfully) unpack fields into function. To avoid that now and in the future, we could either wrap the init response in an `init` object as in #2823 bit this would break all existing plugins.

Reply:
Yeah, that argument unpacking was a bit of a brainfart on my end, but it has been working nicely so far. We should just define a minimal set of top-level objects that need to be passed in and then add new stuff to those. That also makes this semantically clean.

Reply:
Ok, i'll add `startup`to `configuration` then